### PR TITLE
Fix rendering the avatar on entry

### LINKED ISSUE #50: RTCDataChannel not connecting upon room entry
RTCDataChannel should connect once upon entering the room. Instead, onopen called twice after the input is selected.

### PR DESCRIPTION
**Pull request by siberowl**
## Which Issue was this PR made for?

- Closes #50 

## What is within the scope with this PR
Fixing the bug where the avatar wasn't being rendered until an audio input was selected.

## What is NOT in the scope with this PR
Fixing the bug where onopen is called twice.

### Related Issue
- #68 
- #65

### CODE REVIEW

File: client/src/pages/RoomPage.tsx [hunk 1]
```diff
@@ -57,14 +63,18 @@ function RoomPage({ name }: { name: string }): JSX.Element {
...
   // open all listeners on render
   useEffect(() => {
     notifyAndRequestNetworkInfo(socket, name);
-    openJoinListener(socket, onJoin);
+    openJoinListener(socket, onNewJoin);
     openLeaveListener(socket, setAnnouncement, onLeave);
     openRequestUsersListener(socket, setNewUser);
-    openOfferListener(getUsers(), socket);
-    openAnswerListener(getUsers(), socket);
+    openOfferListener(getUsers(), socket, printPack, printPack);
```
Comment (Njima1572) [inline client/src/pages/RoomPage.tsx:76]: (seems dumb

**Comment by Njima1572** [comment]
Does the new user receive positions from existing users upon entry?
Selection_076

**Comment by Njima1572** [comment]
It feels like it should set position when it got first message right???

**Comment by Njima1572** [comment]
https://github.com/Amijakan/spatialchatclone/blob/5f6f35c7716401b7749b33013785dfa5c04994f5/client/src/User.ts#L21

this.selfPosition never gets updated

**Comment by siberowl** [comment]
> https://github.com/Amijakan/spatialchatclone/blob/5f6f35c7716401b7749b33013785dfa5c04994f5/client/src/User.ts#L21
> 
> this.selfPosition never gets updated

It gets updated here is called by RoomPage
https://github.com/Amijakan/spatialchatclone/blob/b208e01e2dbe0beb355bce4ab16733e920b93093/client/src/RTCPeerConnector.ts#L162-L172

File: client/src/pages/RoomPage.tsx (hunk 1 above)
Comment (siberowl) [reply client/src/pages/RoomPage.tsx:76]: True, it was just for debugging purposes so I'll get rid of it.

**Comment by siberowl** [comment]
> Does the new user receive positions from existing users upon entry?
> 
> Left: Existing user
> Right: New user
> 
> Selection_076

Might've slipped my mind. Will test.

**Comment by siberowl** [comment]
> > Does the new user receive positions from existing users upon entry?
> > Left: Existing user
> > Right: New user
> > Selection_076
> 
> For some reason, selfPosition is being stuck at [0, 0] even after updating when initializing the user.
> Output:
> 
> ```
> RoomPage.tsx:84 (2) [145, 163]
> RoomPage.tsx:84 (2) [148, 165]
> RoomPage.tsx:84 (2) [149, 165]
> RoomPage.tsx:58 (2) [0, 0]
> ```
> 
> https://github.com/Amijakan/spatialchatclone/blob/d6f85e7ed70a2e3d02a09d05bbb9ed7830d2406c/client/src/pages/RoomPage.tsx#L55-L61
> 
> https://github.com/Amijakan/spatialchatclone/blob/d6f85e7ed70a2e3d02a09d05bbb9ed7830d2406c/client/src/pages/RoomPage.tsx#L83-L86
> 
> Also need to do this issue once fixed.
> 
> * [[Feature]: Add User class constructor parameters #78](https://github.com/Amijakan/spatialchatclone/issues/78)

The issue was stale closure! Learned something new and very important.
https://dilshankelsen.com/understanding-stale-closures-in-javascript/

**Comment by Njima1572** [comment]
Noooicee

I guess everything that people gets stuck on has a name for the problem.

**Pull request merged**